### PR TITLE
Improve data persistence documentation

### PR DESCRIPTION
**Pull request by mstanleyjones**
Add new specialized topics about volumes, bind mounts, and tmpfs mounts for data persistence

Ready for review!

### CODE REVIEW

File: engine/admin/volumes/bind-mounts.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+directory's contents.
+
+The file or directory does not need to exist on the Docker host already. It is
+created on demand if it does not yet exist. Bind mounts are very performant, but
+they rely on the host machine's filesystem having a specific directory structure
+available. If you are developing new Docker applications, consider using [named
+volumes](volumes.md) instead. You can't use Docker CLI commands to directly
+manage bind mounts.
+
+## Choosing the -v or --mount flag
```
Comment (joaofnfernandes) [inline engine/admin/volumes/bind-mounts.md]: Our UX research into the CLI shows that the `--mount` syntax is easier to learn and use. With `--volume` even people using Docker on a daily basis had to go to the docs to check what was the order of the arguments.

So I think we could do new users a favor by deemphasizing `-v`. Also, if I'm a new user I don't know what I don't know, so at this point I don't have enough knowledge to make an informed decision between `--mount` or `--volume`, so you could give me a very opinionated way of using volumes/bind-mounts if that way would save me time in the long run.

So consider organizing this article slightly differently:
1. What are bind-mounts
2. Start a container with a bind mount
3. Mount into a non-empty directory
4. Use a read-only bind-mount
5. Configure bind propagation
...
x. Differences between `--mount` and `--volume`

File: engine/admin/volumes/tmpfs.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+---
+
+[Volumes](volumes.md) and [bind mounts](bind-mounts.md) are mounted into the
+container's filesystem by default, and their contents are stored on on the host
+machine.
+
+There may be cases where you do not want to store a container's data on the host
+machine, but you also don't want to write the data into the container's writable
+layer, for performance or security reasons, or if the data relates to
+non-persistent application state.
```
Comment (joaofnfernandes) [inline engine/admin/volumes/tmpfs.md]: I think this example is a bit generic so users might have some trouble understanding the usefulness of tmpfs mounts. The example you have in the index.md page about passwords or sensitive data is so much better, so consider using it here.

File: engine/admin/volumes/tmpfs.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+## Use a tmpfs mount in a container
+
+To use a `tmpfs` mount in a container, use the `--tmpfs` flag, or use the
+`--mount` flag with `type=tmpfs` and `destination` options. There is no
+`source` for `tmpfs` mounts. The following example creates a `tmpfs` mount at
+`/app` in a Nginx container. The first example uses the `--tmpfs` flag and the
+second uses the `--mount` flag.
+
+```bash
```
Comment (joaofnfernandes) [inline engine/admin/volumes/tmpfs.md]: I know I'm being boring by now but can we switch the order of these examples to emphasize the use of `--mount`?

File: engine/admin/volumes/volumes.md [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+containers using it, and the volume's contents exist outside the lifecycle of a
+given container.
+
+If your container generates non-persistent state data, consider using a
+[tmpfs mount](tmpfs.md) to avoid storing the data anywhere permanently, and to
+increase the container's performance by avoiding writing into the container's
+writable layer.
+
+Volumes use `rprivate` bind propagation, and bind propagation is not
+configurable for volumes.
```
Comment (joaofnfernandes) [inline engine/admin/volumes/volumes.md]: Maybe include a follow-up sentence to explain to non-Linux geeks what this means.

Comment (mstanleyjones) [inline engine/admin/volumes/volumes.md]: If you need this you are already a Linux geek. I've already explained that it's an advanced option. I don't think I should spend more time on it.

**Comment by mstanleyjones** [comment]
Addressed all feedback, tab-ified examples, updated the node title from "Storage Guide" to "Manage application data". Please re-review.

File: engine/admin/volumes/index.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  Volumes also support the use of _volume drivers_, which allow you to store
+  your data on remote hosts or cloud providers, among other possibilities.
+
+- **[tmpfs mounts](tmpfs.md)**: A `tmpfs` mount is not persisted on disk, either
+  on the Docker host or within a container. It can be used by a container during
+  the lifetime of the container, to store non-persistent state or sensitive
+  information. For instance, internally, swarm services use `tmpfs` mounts to
+  mount [secrets](/engine/swarm/secrets.md) into a service's containers.
+
+Bind mounts and volumes can both mounted into containers using the `-v` or
```
Comment (mstanleyjones) [inline engine/admin/volumes/index.md]: Done

File: engine/admin/volumes/bind-mounts.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+Docker's storage directory on the host machine, and Docker manages that
+directory's contents.
+
+The file or directory does not need to exist on the Docker host already. It is
+created on demand if it does not yet exist. Bind mounts are very performant, but
+they rely on the host machine's filesystem having a specific directory structure
+available. If you are developing new Docker applications, consider using [named
+volumes](volumes.md) instead. You can't use Docker CLI commands to directly
+manage bind mounts.
+
```
Comment (mstanleyjones) [inline engine/admin/volumes/bind-mounts.md]: Done

File: engine/admin/volumes/bind-mounts.md (hunk 1 above)
Comment (mstanleyjones) [inline engine/admin/volumes/bind-mounts.md]: Done but in a slightly different way.

**Comment by mstanleyjones** [comment]
Fixed tabs, rebased on `master` so we can get this in ASAP.

**Comment by mstanleyjones** [comment]
I'm pretty happy with this so going to merge it. @sarahpark we can add that graphic when it's ready. :)

**Pull request merged**